### ISSUE TITLE
Open Beta v7.7 | Please help testing and hardening the upcoming release

### ISSUE THREAD

**Issue by MichaIng** [MichaIng/DietPi#4820]
RC version | v7.7.2
---------- | -------
v7.7.0 PR | #4819
v7.7.1 PR | #4829
v7.7.2 PR | #4839
v7.7.3 PR | #4849 (4819, 4829, 4839 and 4849, what a coincidence 😄)
Raw changelog | https://github.com/MichaIng/DietPi/blob/beta/CHANGELOG.txt
Code changes | https://github.com/MichaIng/DietPi/compare/master...beta
Release date | **2021-10-16**

#### Related/solved issues: https://github.com/MichaIng/DietPi/issues?q=is%3Aissue+milestone%3Av7.7
____
### Beta v7.7.3
_(2021-10-16)_

#### Removed software
- CouchPotato | Sadly, the CouchPotato project is not maintained anymore and has been abandoned. In turn we removed it from DietPi. The instance installed on your system will remain, but it cannot be installed, reinstalled or uninstalled anymore. We recommend to migrate to an alternative project, like Radarr, which can be found in DietPi-Software was well. Manual CouchPotato uninstall instructions: https://github.com/MichaIng/DietPi/issues/4323#issuecomment-927128724

#### Changes
- General | The /boot/dietpi/func/obtain_network_details script has been removed, including the related /run/dietpi/.network file to obtain network details. All uses of these files have been replaced with the new DietPi-Globals G_GET_NET function (see below).
- DietPi-Globals | A new global function G_GET_NET has been added to print network interface details. Most importantly it prints info for the main interface, by following the priorities of /boot/dietpi/func/obtain_network_details: default gateway => state UP => IP assigned, but allows to additionally filter by IP family, type, interface name or print the default gateway explicitly. It aims to be a replacement for /boot/dietpi/func/obtain_network_details with more flexibility and to allow deriving always up-to-date interface info instead of depending on the correctness of a cache file.
- DietPi-Globals | G_GET_WAN_IP: We use our own GEO IP service now to show the systems WAN IP and location in the DietPi banner and DietPi-VPN. When Pi-hole was used, with a previous update, "freegeoip.app" was added to Pi-hole's whitlist, which is now not required anymore. You may hence remove that entry from the whitelist.
- DietPi-Globals | The global functions G_DEV_1 and G_DEV_BENCH have been removed, which did exist for testing and development only but are not used in our current workflows.
- DietPi-TimeSync | Use the same flag file that systemd-timesyncd itself uses since Buster, to skip an additional service restart and sync when it was done already.
- DietPi-TimeSync | When our oneshot modes (boot only, hourly, daily) are selected, systemd-timesyncd is now "enabled" to be started by systemd earlier at boot, instead of on our script call. Especially since both now share the same flag file (on Buster and above), this has a chance to prevent an additional service restart if the time sync has finished already when PostBoot is reached.
- DietPi-Boot | This script and service has been removed: Waiting for network is now done via DietPi-PostBoot "After=network-online.target", time sync is done in DietPi-PostBoot, but in background (mostly not required for service starts) and pre-installed image stage handling is as well done in PostBoot now.
- DietPi-Update | A network connection and time sync check is now done before checking for updates, similar to how DietPi-Software does it on installs.
- DietPi-Login | The DietPi banner on login won't be shown anymore if ~/.hushlogin exists, which is a common method to prevent the shell from printing /etc/motd on login and should hence be respected for the DietPi banner as well. Many thanks to @dnknth for doing this suggestion: https://github.com/MichaIng/DietPi/issues/4786
- DietPi-JustBoom | Added the ability to enforce an output channel count, or to not enforce an audio format value to preserve the input stream format or leave conversion up to ALSA, which is now the default when resetting settings. Similarly the audio output buffer can now be unset to keep the MPD default. Generally, if not required for a specific reason, it is recommended to not convert the audio stream and keep these settings unchanged/default.
- DietPi-Software | Deluge: Logging is not done to /var/log/deluged/ anymore but to journal instead, accessible via "journalctl -u deluged -u deluge-web". This change only affects new installs and reinstalls of Deluge.
- DietPi-Software | Deluge: On fresh installs, the web interface is now accessible as expected with the chosen global software password, stored hashed with a fresh random salt. Previously the password was hardcoded to "dietpi".
- DietPi-Software | Kodi: On Debian Bullseye, with Kodi 19, GBM support is present by default, which means that Kodi can be started without a wrapping X server. This is now done by default when starting Kodi outside of a desktop session, including the dietpi-autostart option. This also means that an X server is not installed anymore as a dependency of Kodi, but only as a dependency of a deskop environment.
- DietPi-Software | Kodi: It can now be installed on all devices. In some cases, video playback performance may be bad, depending on the GPU, whether good drivers are available, and on the video quality, of course. However, it should be our users who evaluate whether it's sufficient or not, instead of us. With Debian Bullseye, new Mesa drivers and Kodi 19 started via GBM, performance should be much better than it was with older Debian/package versions.
- DietPi-Software | File Browser: The default network port has been changed to 8084 to resolve a conflict with HTPC Manager. This only affects new File Browser installs. Many thanks to `@KamikazeePL` for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?t=9507

#### Fixes
- General | Since the Armbian repository router does not reliably preserves HTTPS on redirects yet, APT by times fails when detecting a downgrade from HTTPS to HTTP. We hence change the armbian.list to use plain HTTP until the issues with the router have been resolved.
- General | Worked around an issue on Debian Stretch where "systemctl enable/disable --now" does not start/stop the service in certain circumstances. This is solved within our error handler "G_EXEC", hence when manually calling systemctl you may still face this: https://github.com/MichaIng/DietPi/issues/4815
- General | Applied a workaround on Bullseye systems with older Linux versions (v4.14 and below) which do not sufficiently support the new "unified cgroup hierarchy" (a.k.a. cgroups-v2). Since the newer systemd tries to use it automatically, Docker and similar software which make use of cgroups fail. For devices with known boot configuration file, the kernel command line arguments are applied to force the legacy cgroups hierarchy usage.
- DietPi-Backup | Resolved an issue where clearing the PATH cache via "hash" command did not work as of a wrong command line argument: https://github.com/MichaIng/DietPi/issues/4800
- DietPi-LetsEncrypt | Resolved an issue where the script failed when ownCloud or Nextcloud are were installed. Many thanks to @billouetaudrey for reporting this issue: https://github.com/MichaIng/DietPi/issues/4752
- DietPi-JustBoom | Resolved an issue where applying some MPD settings did not work. Many thanks to `@elevader` for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?t=9426
- DietPi-Config | Resolved an issue where the WiFi connection state could have been obtained falsely as accidentally the Ethernet interface index was used to derive it.
- DietPi-Config | Resolved an issue on NanoPi NEO (and likely other Allwinner H3 SBCs) where selecting a sound card failed as an invalid control was tried to be set. Many thanks to @VS-X for reporting this issue: https://github.com/MichaIng/DietPi/issues/4833
- DietPi-Software | Resolved an issue where software services failed with a cryptic error message, when an expected directory was not present. This was especially reported with Sonarr and Radarr, if their log directory was missing for some reason. When directories are missing, which are explicitly listed to be read-writeable within the systemd service, systemd prints "Failed at step NAMESPACE spawning", while Sonarr and Radarr themselves would print a clearer error message about the missing log directory. Many thanks to `@stevewitz` for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?t=9463
- DietPi-Software | Lighttpd: Resolved an issue where the upgrade from Buster to Bullseye, following our guide, fails if HTTPS was enabled via DietPi-LetsEncrypt before. Many thanks to `@fhals` for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?t=9477
- DietPi-Software | Kodi: Resolved an issue on RPi ARMv8/64-bit systems where Kodi fails to start when it was installed without a desktop. Many thanks to `@Klola` for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?p=38079#p38079
- DietPi-Software | Deluge: Resolved an issue on Bullseye where the web interface service did not start as a new command line flag "-d" is required to keep it in foregound. Many thanks to @quyentruong for reporting this issue: https://github.com/MichaIng/DietPi/issues/4785
- DietPi-Software | Home Assistant: The Python version compiled with Home Assistant has been bumped to v3.9.7, which resolves and issue with installs on 32-bit ARM systems. Many thanks to @Przemek for reporting this issue: https://github.com/MichaIng/DietPi/issues/4372#issuecomment-936656595
- DietPi-Software | Home Assistant: Resolved an issue where Home Assistant did not start on ARM systems due to newly required runtime libraries.
- DietPi-Software | Chromium: Resolved an issue where the autostart option didn't work if Chromium was installed without a desktop. Many thanks to `@jowelboy` for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?t=9531
- DietPi-Software | Chromium: Resolved an issue on RPi where starting Chromium failed if no desktop was installed, due to a missing dependency. Many thanks to @Loader23 for reporting this issue: https://github.com/MichaIng/DietPi/issues/4782
- DietPi-Software | X.Org X Server: Resolved an issue with Odroid N2 and C4 models where the installation failed because of a typo. Many thanks to @wiml for reporting this issue: https://github.com/MichaIng/DietPi/issues/4830
- DietPi-Software | Airsonic: Since the project has been archived and does not support Java 17, it has been disabled on Bullseye. We're watching a fork (https://github.com/airsonic-advanced/airsonic-advanced) which is actively developed and where at least the web interface works with Java 17. Playing audio however failed on local tests, hence we'll wait until it becomes more stable to be a drop-in replacement for Airsonic in general and supported on Bullseye with Java 17 as well. Many thanks to @Andaloup for reporting this issue: https://github.com/MichaIng/DietPi/issues/4847
- DietPi-Software | FreshRSS: Resolved an issue where on reinstalls nested /opt/FreshRSS/FreshRSS-master and /opt/FreshRSS/p/p were created. Since FreshRSS has an internal updater, reinstalls won't download and install the new version as long as /opt/FreshRSS is present already. The nested directory and link is removed on next DietPi update, when present. Many thanks to @kinoushe for reporting this issue: https://github.com/MichaIng/DietPi/issues/4775

**Comment by MichaIng** [comment]
`vm.swappiness=0` does not disable swap usage but reduces it's usage to a minimum. This is mostly reasonable with an SD card as system drive where you want to avoid any possible disk I/O, with an SSD not so much, depending a bit on usage in general:
- Generally of course the swap space is much slower than the RAM, so keeping things in RAM instead of writing them to disk generally, on first view, increases performance, while the letter should only be considered a worst case fallback, when there is simply no left space in RAM, to avoid OOM kills. `vm.swappiness=0` behaves moreless like that, by only swapping data from RAM when physical RAM is mostly full.
- But due to Linux filesystem caching sometimes it makes sense to swap rarely accessed RAM data earlier. E.g. you may run a Java application which usually reserves several hundred MiBs in RAM to store the full runtime and for heap data, while the actually only a little part of Java classes and heap are (regularly) used, and the reserved memory is mostly a waste of RAM space. Now when the RAM is mostly full, with `vm.swappiness=0` foreground process data is still kept preferably in physical RAM, so that there is few RAM available and used for the filesystem cache. So another process which e.g. reads often accessed files from disk will be slower, because that disk files are not cached to RAM while the useless Java data is reserving it.

_Sorry for Java bashing here, but for low memory devices this is a real problem 😄._

**Comment by billouetaudrey** [comment]
By default, it's 1 right ?

**Comment by MichaIng** [comment]
Ah lol, whoops, `vm.swappiness=0` DOES disable swap usage completely, `vm.swappiness=1` is minimum usage (and default on DietPi via `/etc/sysctl.d/dietpi.conf`). So that is what may be tested.

**Comment by MichaIng** [comment]
@billouetaudrey 
Without a swap file your memory usage is close to max, so a swap file definitely makes sense. As you can see, which higher swappiness the used swap space is higher and the used RAM space lower, as expected. What is missing is the `buff/cache` info to know whether the additional available RAM space has been effectively used for filesystem caching or not. E.g. here on my server:
```
# free -m
               total        used        free      shared  buff/cache   available
Mem:             972         385         150          93         437         478
Swap:           4095           1        4094
```
Means from 972 MiB RAM, 385 MiB are actively used and another 437 MiB were just filled with regularly accessed files from the filesystem, to speed up further accesses. The remaining 150 MiB in my case of completely unused, and obviously this didn't change significantly as the swap file was used with only 1 MiB. Since the filesystem cache is not strictly required, it can be freed quickly to make additional RAM available, if required, hence the higher "available" RAM of 478 MiB. But when accessing those files then at a later time, they need to be read from disk, hence it will take longer. In my case however a higher swappiness would have zero effect as of the completely unused 150 MiB physical RAM space. I only have the swap file because I can and it doesn't hurt as its on a 3 TiB 15% used external drive.

**Comment by MichaIng** [comment]
#### Beta v7.7.2 has been released: #4839

**Comment by MichaIng** [comment]
#### DietPi v7.7 has just been released, many thanks to all testers ❤️!
